### PR TITLE
ENH add helper to store metadata in parquet files

### PR DESCRIPTION
**Pull request by tomMoral**
### Checks before merging PR
- [x] ~added documentation for any new feature~
- [x] added unit test
- [x] edited the [what's new](../../whatsnew.rst) (if applicable)

### CODE REVIEW

File: benchopt/utils/parquet.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
+
```
Comment (Badr-MOUFAD) [inline benchopt/utils/parquet.py:1]:
```suggestion
```

File: benchopt/utils/parquet.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import json
+import pyarrow as pa
+import pyarrow.parquet as pq
+
+JSON_KEY = b"_meta_json"
+
+
+def to_parquet(df, path, metadata=None):
+    """Write a pandas.DataFrame in a parquet file, with optional metadata.
```
Comment (Badr-MOUFAD) [inline benchopt/utils/parquet.py:10]:
```suggestion
    """Write a ``pandas.DataFrame`` in a parquet file, with optional metadata.
```

File: benchopt/utils/parquet.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+def update_metadata(path, metadata):
+    """Update metadata in an existing parquet file.
+
+    Parameters
+    ----------
+    path: str | Path
+        Path of the parquet file to update.
+    metadata: dict or None
+        Metadata to store in the parquet file. This metadata should be
+        serializable with json.
```
Comment (Badr-MOUFAD) [inline benchopt/utils/parquet.py:40]:
```suggestion
        serializable with ``json``.
```

File: benchopt/utils/parquet.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    }
+    metadata_ = json.loads(new_metadata.get(JSON_KEY, b"{}").decode("utf-8"))
+    metadata_.update(metadata)
+    new_metadata[JSON_KEY] = json.dumps(metadata_).encode("utf-8")
+    table = table.replace_schema_metadata(new_metadata)
+    pq.write_table(table, path)
+
+
+def get_metadata(path):
+    """Retrieves metadata embeded using `to_parquet` function.
```
Comment (Badr-MOUFAD) [inline benchopt/utils/parquet.py:54]:
```suggestion
    """Retrieve metadata embedded using the ``to_parquet`` function.
```

File: benchopt/utils/tests/test_parquet.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+
+def test_parquet_metadata():
+    df = pd.DataFrame({
+        'a': range(4),
+        'b': list('abcd'),
+        'c': [.1, .2, .3, .4],
+    })
+
+    with tempfile.NamedTemporaryFile("rb+", suffix=".pq") as f:
```
Comment (Badr-MOUFAD) [inline benchopt/utils/tests/test_parquet.py:17]: Perhaps I'm missing some context, shouldn't the extension be `.parquet`?

File: benchopt/utils/tests/test_parquet.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+        path = f.name
+        metadata = {'test': 'info', 'plot_configs': [{}, {}, {}]}
+        to_parquet(df, path, metadata)
+
+        # Make sure the data is readable
+        df_new = pd.read_parquet(path)
+        assert (df_new == df).all().all()
```
Comment (Badr-MOUFAD) [inline benchopt/utils/tests/test_parquet.py:25]:
```suggestion
        assert df.equals(df_new)
```

I would remove it as the same check is done below

File: benchopt/utils/tests/test_parquet.py [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+        # Check that the metadata can be retrieved
+        assert json.dumps(get_metadata(path)) == json.dumps(metadata)
+
+        meta_update = {'test': 'ofni', 'a': 'b'}
+        metadata.update(meta_update)
+        update_metadata(path, meta_update)
+
+        # Make sure the data did not change
+        df_new = pd.read_parquet(path)
+        assert (df_new == df).all().all()
```
Comment (Badr-MOUFAD) [inline benchopt/utils/tests/test_parquet.py:36]:
```suggestion
        assert df.equals(df_new)
```

File: benchopt/utils/parquet.py (hunk 3 above)
Comment (tomMoral) [reply benchopt/utils/parquet.py:40]: This is the protocol name Json, not the module so I would not put quotes here.

File: benchopt/utils/tests/test_parquet.py (hunk 6 above)
Comment (tomMoral) [reply benchopt/utils/tests/test_parquet.py:25]: The test needs to be done twice to verify updating the metadata did not change the file content

**Pull request merged**